secator/hooks/json.py

```
@@ -0,0 +1,157 @@
+	for output_type in FINDING_TYPES:
+		type_name = output_type.get_name()
+		results[type_name] = [
+			item.toDict() if hasattr(item, 'toDict') else item
+			for item in self.results
+			if getattr(item, '_type', None) == type_name
+		]
+	try:
+		with open(report_path, 'w') as f:
+			json.dump({'info': info, 'results': results}, f, cls=DataclassEncoder, indent=2)
+		debug(f'Wrote final JSON report to {report_path}', sub='hooks.json')
+	except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)